File: Frontend_Drizzle/src/app/State/service/product.service.ts
import {Injectable} from '@angular/core';
import {BASE_API_URL} from "../../config/api";
import {HttpClient, HttpParams} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  private apiUrl = BASE_API_URL + "/api";

  constructor(
    private http: HttpClient,
  ) {
  }

  findProductByCategory(reqData: any) {
    const {
      colors,
      sizes,
      minPrice,
      maxPrice,
      minDiscount,
      category,
      stock,
      sort,
      pageNumber,
      pageSize,
    } = reqData;

    let params = new HttpParams()
      .set("color", colors)
      .set("size", sizes)
      .set("minPrice", minPrice)
      .set("maxPrice", maxPrice)
      .set("minDiscount", minDiscount)
      .set("category", category)
      .set("category", stock)
      .set("sort", sort)
      .set("pageNumber", pageNumber)
      .set("pageSize", pageSize);

    return this.http.get(`${this.apiUrl}/products`, {params});
  }

  findProductById(productId: any) {
    return this.http.get(`${this.apiUrl}/products/id/${productId}`, {});
  }
}

File: Drizzle_backend/src/main/java/com/gyalbu/drizzle_backend/service/ProductService.java
package com.gyalbu.drizzle_backend.service;

import com.gyalbu.drizzle_backend.entity.Product;
import com.gyalbu.drizzle_backend.exception.ProductException;
import com.gyalbu.drizzle_backend.resources.request.CreateProductRequest;
import org.springframework.data.domain.Page;

import java.util.List;

public interface ProductService {

    Product createProduct(CreateProductRequest request) throws ProductException;

    String deleteProduct(Long productId) throws ProductException;

    Product updateProduct(Long productId, Product request) throws ProductException;

    Product findProductById(Long productId) throws ProductException;

    List<Product> findProductByCategory(String category);

    Page<Product> getAllProduct(String category,
                                List<String> colors,
                                List<String> sizes,
                                Integer minPrice,
                                Integer maxPrice,
                                Integer minDiscount,
                                String sort,
                                String stock,
                                Integer pageNumber,
                                Integer pageSize);

    List<Product> findAllProducts();
}
